chef/cookbooks/ohai/files/default/plugins/linux/block_device_extended.rb

```diff
@@ -25,6 +25,22 @@
         File.open("/sys/block/#{dir}/queue/#{check}") { |f| block[dir][check] = f.read_nonblock(1024).strip }
       end
     end
+
+    # correct that the aacraid driver always sets the removable flag
+    if block[dir]["removable"] == "1"
+      path = File.readlink("/sys/block/#{dir}")
+      path = File.realpath(path, "/sys/block/").split("/")
```

Comment:
Do you really need that `readlink` here? I think `path = File.realpath("/sys/block/#{dir}").split("/")` should work...